src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataArchiveValidationService.cs

```diff
@@ -72,5 +85,37 @@ private async Task<bool> IsZipFile(IFormFile file)
                    )
                    && _fileTypeService.HasMatchingEncodingType(file, ZipEncodingTypes);
         }
+
+        private static Either<ActionResult, Unit> ValidateFilenameLengths(
+            int zipFilenameLength,
```

Comment:
I'd be inclined to make one instance of `ValidateFilenameLengths` and make `zipFilenameLength` an optional argument that defaults to null, rather than having two instances of it. Then we can also have one instance of `MaxFilenameSize` and avoid any mishaps with someone changing one without realising they need to change the other.

It's possible there is something I don't understand here with one `ValidateFilenameLengths` method returning a list of errors while the other only returns a single error.

If we do only have one instance of `ValidateFilenameLengths`, there is the question of where it should live. I'm not fussed - in either service is fine by me - but others might have an opinion on it. Could argue it should be in another service so it can be unit tested separately - that seems like overkill to me.

Reply:
Since the zip file is uploaded from a single form field, it can only display a single error message - there's currently no logic to handle a list on the front-end (e.g. combine and comma separate) - whereas the individual files can return an error message for each field.

I tried a few variations to combine the logic into a single method, but they weren't as readable and decided against it.